flow_runner.go

```diff
@@ -142,37 +147,121 @@ func (f *flowRunner) runTask(ctx context.Context, task Task) bool {
 		return true
 	}
 
-	w := f.output
-	if !f.verbose {
-		w = &strings.Builder{}
+	paramValues := make(map[string]Value)
+	for _, param := range task.Parameters {
+		paramValues[param.Name()] = f.paramValues[param.Name()]
+	}
+	if f.verbose != nil {
+		paramValues[f.verbose.Name()] = f.paramValues[f.verbose.Name()]
 	}
```

Comment:
If I understand correctly, becasue of this code a `Command` can use the `verbose` param even though it was not explicitly assigned in `Task.Parameters`. Am I correct?
If so then I am not sure if it is a good idea.

Reply:
This was necessary because the parameter check is strict in testing whether the parameter is assigned to the command.
Tests did fail because the command has the verbose parameter not explicitly assigned. (After all, commands are not allowed to make use of parameters that weren't assigned).
A similar check is for testing whether parameters are unused.
In short, the "magic" that happens is that this parameter is a global one, assigned to all commands, as they make use of it implicitly when using output.

Reply:
Let's keep as it is for now 😉

Reply:
What do you think about: https://github.com/dertseha/taskflow/pull/2 ?

Reply:
This approach is also fine with me. During implementation I had a short idea to consider "tasks within tasks" - so Taskflow would be running a "global" task (which would have the verbose parameter registered), and all user-tasks were sub-tasks of that one.
Though, introducing the concept of sub-tasks sounds over-engineered.
So, I'm happy with any approach to solve the issue about global parameters.

Reply:
(PR accepted)